File: C++ Projects/Quiz/questionmc.cpp
//-------------------------
//class implentation(.cpp)
//author: Noah Funderburgh
//File:questionmc.cpp
//-------------------------
#include "question.h"
#include "questionmc.h"

using namespace std;
//------------------------
questionMC::~questionMC(){

}
//------------------------
questionMC::questionMC(){

}
//------------------------
void questionMC::multiChoice(const string &choices){
    int size = count(choices.begin(),choices.end(),':');
        stringstream ss(choices);
        string token;
        for(int i = 0; i<=size;i++){
            getline(ss,token,':');
            cout <<"   " <<  token << endl;
        }

}
//------------------------
void questionMC::showQuestion(){
    int size = count(questionText.begin(),questionText.end(),':');
    char letter = 'A';
        stringstream ss(questionText);
        string token;
        for(int i = 0; i<=size;i++){
            getline(ss,token,':');
            if(i ==0){
                cout << token << "?" << endl;
            }
            else{
            cout <<"   (" << letter <<") " <<  token << endl;
            letter++;
            }
        }
}
//------------------------
void questionMC::showAnswer(){
    cout << answerText << endl;
}
//------------------------
bool questionMC::checkAnswer(string answer){
    string answerTextLower = answerText;
    for(int i = 0;i<=answer.length();i++){
        if(answer[i] >= 65 && answer[i]<=92){
            answer[i]=answer[i]+32;
        }
    }

    for(int i = 0;i<=answerTextLower.length();i++){
        if(answerTextLower[i] >= 65 && answerTextLower[i]<=92){
            answerTextLower[i]=answerTextLower[i]+32;
        }
    }

    if(answer.at(0) == answerTextLower.at(0)){
        correct = true;
        return true;
    }
    else{
        correct = false;
        return false;
    }
}
//------------------------

File: Java Projects/Quiz/Questionmc.java

import java.util.Arrays;

/**
* Implements classes need to add a multiple choice question
* Functionality to our Quiz
*
* @author  Noah Funderburgh
* @version 1.0
* @since   3/3/2020
*/

public class Questionmc extends Question {
	/**
	 * This functions will show the answer when calling the function for multiple choice quesitons.
   	 */
	void showQuestion() {
		char charLookingFor = ':';
		int count = 0;
		char letter = 'A';
		String[] tokens = questionText.split("\\:");
		for (int i = 0; i < questionText.length(); i++) {
			if (questionText.charAt(i) == charLookingFor) {
				count++;
			}
		}
		count++;

		for (int i = 0; i < count; i++) {
			if (i == 0) {
				System.out.println(tokens[0] + "?");
			} else {
				System.out.println("    (" + letter + ") " + tokens[i]);
				letter++;
			}
		}
	}

//-------------------------------------------------------
	/**
	 * This function checks the if userAnswer is equal to Answer by making both strings lowercase
	 * and then comparing the two to check if there equal.
	 * 
	 * @param userAnswer answer thats input by the user in the quiz
	 * @return boolean returns true if there equal else it returns false.
   	 */
	boolean checkAnswer(String userAnswer) {
		userAnswer = userAnswer.toLowerCase();
		answerText = answerText.toLowerCase();
		
		char userArray[] = userAnswer.toCharArray();
		Arrays.sort(userArray);
		userAnswer = String.valueOf(userArray);
		
		String answer = answerText;
		char answerArray[] = answer.toCharArray();
		Arrays.sort(answerArray);
		answer = String.valueOf(answerArray);
	
		
		if(answer.equals(userAnswer)) {
			return true;
		}
		else {
			return false;
		}
	}
}

//-------------------------------------------------------